src/complex/Utilities/ParallelDataAlgorithm.hpp

```diff
@@ -26,7 +26,12 @@ class COMPLEX_EXPORT ParallelDataAlgorithm
   using RangeType = Range;
 
   ParallelDataAlgorithm();
-  virtual ~ParallelDataAlgorithm();
+  ~ParallelDataAlgorithm();
+
+  ParallelDataAlgorithm(const ParallelDataAlgorithm&) = delete;
+  ParallelDataAlgorithm(ParallelDataAlgorithm&&) noexcept = delete;
+  ParallelDataAlgorithm& operator=(const ParallelDataAlgorithm&) = delete;
+  ParallelDataAlgorithm& operator=(ParallelDataAlgorithm&&) noexcept = delete;
```

Comment:
I think all of these can be defaulted. Being copyable/movable should be fine for these classes. Also would we want to rename `ParallelDataAlgorithm` to `ParallelData1DAlgorithm` to be consistent with the other classes?